### PR TITLE
Sorting clients by name

### LINKED ISSUE #35: As an insurance agent, I want to sort my clients, so that I can see my clients in a more organized way.

### PR DESCRIPTION
**Pull request by NBH99**
Add sort command which takes in user input "sort DIRECTION" where DIRECTION can be "-a" for ascending or "-d" for descending.
JSON file will be affected by the sorting but I think if a person want to sort then he probably want the file to be sorted anyway.
You can clone https://github.com/NBH99/tp/tree/Sort to see how it works.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        Comparator<Person> comparator = new PersonNameComparator();
+        if (direction.equals("/d")) {
+            comparator = comparator.reversed();
+        }
+        model.updateSortedPersonList(comparator);
+        if (direction.equals("/a")) {
```
Comment (swayongshen) [inline src/main/java/seedu/address/logic/commands/SortCommand.java]: Would it be better to have a static constant for each direction?

File: src/main/java/seedu/address/logic/parser/SortCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+     */
+    public SortCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        String direction = trimmedArgs.split("\\s+")[0];
+        if (!direction.equals("/a") && !direction.equals("/d")) {
```
Comment (swayongshen) [inline src/main/java/seedu/address/logic/parser/SortCommandParser.java]: Would it be better to have static constant for each direction?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -148,6 +149,19 @@ Examples:
 * `find Betsy` followed by `delete 1` deletes the 1st person in the results of the `find` command.
 
 
+### Sorting the list of clients : `sort`
+
+Sort the list of clients in ClientBook by their names in alphabetical order.
+
+Format: `sort DIRECTION`
+
+* Sort the list of clients according the specified `DIRECTION`.
```
Comment (swayongshen) [inline docs/UserGuide.md]: Is using "sorts" instead of "sort" better as it follows the convention used by the other commands?

**Review by swayongshen — COMMENTED** [review]
Very good PR, I like that you updated the various documentation which reminded me to do it for my PR :). I did a light testing of the feature on my PC which worked great. Just some minor improvements and you can go ahead to merge it :+1:

File: docs/UserGuide.md (hunk 3 above)
Comment (NBH99) [reply docs/UserGuide.md]: Noted thanks for the review

**Review by swayongshen — APPROVED** [review]
lgtm

**Pull request merged**